src/python.rs

```diff
@@ -345,6 +437,96 @@ impl Backend {
+				debug!("Checking identifier '{}' at offset {}", identifier, offset);
+
+				// Check if this identifier is an imported name or alias
+				if let Some(import_info) = imports.get(identifier.as_ref()) {
+					debug!("Found import info for '{}': {:?}", identifier, import_info);
+					// Try to resolve the module path to a file
+					if let Some(file_path) = self.index.resolve_py_module(&import_info.module_path) {
+						debug!("Resolved file path: {}", file_path.display());
+						// Try to find the specific class/function definition in the target file
+						if let Ok(target_contents) = std::fs::read(&file_path) {
+							// Look for the class definition
+							let class_name = &import_info.imported_name;
+
+							debug!("Looking for class '{}' in target file", class_name);
+
+							// Use tree-sitter to parse the target file and find the class definition
+							let mut target_parser = Parser::new();
+							target_parser.set_language(&tree_sitter_python::LANGUAGE.into()).unwrap();
+
+							if let Some(target_ast) = target_parser.parse(&target_contents, None) {
+								// Look for class definitions
+								let mut target_cursor = target_ast.walk();
+
+								fn find_class_definition<'a>(cursor: &mut tree_sitter::TreeCursor<'a>, contents: &[u8], class_name: &str) -> Option<tree_sitter::Node<'a>> {
+									if cursor.node().kind() == "class_definition" {
+										// Check if this is the class we're looking for
+										if let Some(name_node) = cursor.node().child_by_field_name("name") {
+											let name = String::from_utf8_lossy(&contents[name_node.byte_range()]);
+											if name == class_name {
+												return Some(name_node);
+											}
+										}
+									}
+
+									if cursor.goto_first_child() {
+										loop {
+											if let Some(found) = find_class_definition(cursor, contents, class_name) {
+												return Some(found);
+											}
+											if !cursor.goto_next_sibling() {
+												break;
+											}
+										}
+										cursor.goto_parent();
+									}
+									None
+								}
+
+								if let Some(class_node) = find_class_definition(&mut target_cursor, &target_contents, class_name) {
+									let range = class_node.range();
+									debug!("Found class '{}' at line {}, col {}", class_name, range.start_point.row, range.start_point.column);
+									return Ok(Some(Location {
+										uri: Uri::from_file_path(file_path).unwrap(),
+										range: Range::new(
+											Position::new(range.start_point.row as u32, range.start_point.column as u32),
+											Position::new(range.end_point.row as u32, range.end_point.column as u32)
+										),
+									}));
+								} else {
+									debug!("Class '{}' not found in target file using tree-sitter", class_name);
+								}
+							} else {
+								debug!("Failed to parse target file with tree-sitter");
+							}
+						} else {
+							debug!("Failed to read target file: {}", file_path.display());
```

Comment:
For deeply nested failure conditions we should wrap this in a helper that returns either `Option<_>` or `Result<Option<_>>`